sdk/include/reactos/dbgbitmap.h

```diff
@@ -0,0 +1,359 @@
+/*
+ * PROJECT:     ReactOS SDK
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Helper functions to debug RTL_BITMAP
+ * COPYRIGHT:   Copyright 2024 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+#pragma once
+
+#include <rtltypes.h>
+#include <rtlfuncs.h>
+
+typedef struct _RTL_BITMAP_DBG
+{
+    union
+    {
+        struct
+        {
+            ULONG SizeOfBitMap;
+            PULONG Buffer;
+        };
+        RTL_BITMAP BitMap;
+    };
+    ULONG NumberOfSetBits;
+    ULONG BitmapHash;
+} RTL_BITMAP_DBG, *PRTL_BITMAP_DBG;
+
+static inline
+ULONG fnv1a_hash(ULONG *data, size_t length)
+{
+    ULONG hash = 2166136261u; // FNV offset basis (a large prime number)
+    for (size_t i = 0; i < length; ++i)
+    {
+        hash ^= data[i];
+        hash *= 16777619u; // FNV prime
+    }
+    return hash;
+}
+
+static inline
+ULONG
+RtlComputeBitmapHashDbg(
+    _In_ PRTL_BITMAP_DBG BitMapHeader)
+{
+    ULONG SizeInBytes = ALIGN_UP_BY(BitMapHeader->BitMap.SizeOfBitMap, 32) / 32;
```

Comment:
This includes up to 31 bits that do not belong to the bitmap, can we be sure they do not change?

Reply:
Probably? I hope so. :)